utils/validations/test_mocks.go

```diff
@@ -122,6 +153,10 @@ func XrayServer(t *testing.T, xrayVersion string) (*httptest.Server, *config.Ser
 				}
 			}
 		}
+		if !xscutils.IsXscXrayInnerService(params.XrayVersion) {
```

Comment:
Why do we need lines 156-159? If we need xscServer we will call `XscServer` function

Reply:
Those lines adds to the Xray mock service the inner API of XSC from the specific version to mock the actual Xray service. we are starting to use the `XrayDetails` and not `XscDetails` since the service is now inside Xray.